### DISCUSSION TITLE
Add regex transformation

### DISCUSSION THREAD

**Discussion by pyrog** [grafana/grafana-infinity-datasource#438] [Ideas]
Hi,

I would like to clean data with regex in a panel and reuse it in another panel to avoid multiples queries and formatting.
It's data source is an HTML centipede_logs_infinity.

Currently, I use **Overrides** and **Value mappings** i.e. to parse a NTRIP agent:
`NTRIP BNC/2.12.18 (LINUX)` become agent=`BNC` and version=`2.12.18 (LINUX)`.

It work fine to display data, but when I reuse this panel I must copy all regex.

Is it possible to add regex transformations directly in infinity-datasource?

**Comment by yesoreyeram** [comment]
Have you tried extract function of UQL? https://sriramajeyam.com/grafana-infinity-datasource/wiki/uql and https://github.com/yesoreyeram/uql/blob/8dab28ca6bdec44a6d1ab8f3ae64832f85af85ee/src/parser/project/project.test.ts#L3-L30 have example.

**Comment by pyrog** [comment]
Didn't know UQL before. It could do the job 😃 

…but seem complicated:
- Need to learn another Query Langage.
- UQL is still in BETA 😟
- My data source is an HTML array

I guess that I must use parse-xml to find the rows and cols but I don't understand clearly how.

<img width="100%" alt="Source webpage" src="https://user-images.githubusercontent.com/2158081/196691251-0fccda97-1140-4dae-9cfe-74aaab35bd20.png">
<img width="100%" alt="centipede_logs_infinity query" 
src="https://user-images.githubusercontent.com/2158081/196692242-cb830538-c50e-44b5-9ce3-96e7137661cb.png">
<img width="75%" alt="Override" src="https://user-images.githubusercontent.com/2158081/196694065-a48b0867-bc30-45e2-9713-ed99a4a7ed6a.png">

**Reply by yesoreyeram** [reply]
oops.. definitely UQL doesn't support HTML parsing yet. So I would suggest to stick with Grafana overrides.

**Reply by pyrog** [reply]
Could be fine if HTML Infinity data source could support expressions in backend.
UQL, javascript like expression e.g. [OpenRefine GREL](https://docs.openrefine.org/manual/grel)… or better, Go expressions 😎

**Comment by pyrog** [comment]
Also, UQL seem written entirely in typescript.

For performance issue, "transformations" could be done once in the backend (in Go language) ?
